smoke/IotEdgeQuickstart/details/IotedgedWindows.cs

```diff
@@ -132,7 +133,37 @@ public async Task Configure(string connectionString, string image, string hostna
             }
         }
 
-        public Task Start() => Task.CompletedTask; // Runtime starts automatically when it's installed
+        public async Task Start()
+        {
+            Console.WriteLine("Starting up iotedge service on Windows");
+
+            // Configured service is not started up automatically in Windows 10 RS4, but should start up in RS5.
+            // Therefore we check if service is not running and start it up explicitly
+            try
+            {
+                ServiceController iotedgeService = ServiceController.GetServices().Single(s => s.ServiceName == "iotedge");
+
+                if (iotedgeService.Status != ServiceControllerStatus.Running)
+                {
+                    using (var cts = new CancellationTokenSource(TimeSpan.FromMinutes(2)))
+                    {
+                        iotedgeService.Start();
+
+                        // Wait for service to become active
+                        do
```

Comment:
Since you're already using `ServiceController`, you should use its `.WaitForStatus()` function rather than looping manually.

Reply:
the loop will also exit if it is more than 2 mins